### PR TITLE
enhancement(codecs): Add enhanced metric tag option to json and text encoders

### LINKED ISSUE #15423: Update codecs that could handle enhanced metric tags
The following codecs decode or encode metrics, and should be audited to ensure they handle enhanced metric tags properly. This may require adding options or even splitting codecs to determine if, for example, the `json` encoder should encode metrics with single values as it currently does or with lists of values.

* [ ] `json` encoder
* [ ] `text` encoder

### PR DESCRIPTION
**Pull request by bruceg**
This change adds the `metric_tag_values` option to the `json` and `text` encoders. This is not being done at the more generic transformer layer, as it creates complications for the `native` and `native_json` codecs which must retain all tag values.

This replaces #15701 
Closes #15423

### CODE REVIEW

File: lib/codecs/src/encoding/format/json.rs [hunk 1]
```diff
@@ -33,12 +45,15 @@ impl JsonSerializerConfig {
 
 /// Serializer that converts an `Event` to bytes using the JSON format.
 #[derive(Debug, Clone)]
-pub struct JsonSerializer;
+pub struct JsonSerializer {
+    #[allow(dead_code)]
```
Comment (neuronull) [inline lib/codecs/src/encoding/format/json.rs:49]: 🤔  maybe fog from time off but, seems like this should be removed?

**Review by neuronull — COMMENTED** [review]
The changes to `src/` look fine, just updated to reflect the API changes.

Attempted to review the stuff in `lib/` , didn't find too much.

File: lib/codecs/src/encoding/format/json.rs (hunk 1 above)
Comment (bruceg) [reply lib/codecs/src/encoding/format/json.rs]: Fixed in 49e470b

**Review by neuronull — APPROVED** [review]
LGTM from integrations perspective.

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**